### PR TITLE
Add options to always approve write and execute operations

### PR DESCRIPTION
**Pull request by stea9499**
## Description
Add setting to always approve write and execute operations.

## Related Issues
N/A

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other

## Screenshots/How Has This Been Tested?

**New settings UI:**

cline_settings

**Example task which doesn't prompt for write or execute permissions:**

task_with_write_or_execute

### CODE REVIEW

**Comment by mbenhamd** [comment]
Thanks @stea9499 ; @saoudrizwan I think a lot of people would like to have these features on the next release! :)

**Comment by FloMaetschke** [comment]
nice!

**Comment by KJ7LNW** [comment]
would like regex for _which_ files or cmds to allow.  Just add `.*` if you really want all.

File: package-lock.json [hunk 1]
```diff
@@ -5366,207 +7538,282 @@
...
-      "dependencies": {
-        "boolbase": "^1.0.0",
-        "css-select": "^5.1.0",
-        "css-what": "^6.1.0",
-        "domelementtype": "^2.3.0",
-        "domhandler": "^5.0.3",
-        "domutils": "^3.0.1"
+    "node_modules/esbuild": {
+      "version": "0.24.0",
+      "resolved": "https://roo-815250993495.d.codeartifact.us-east-1.amazonaws.com/npm/roo-dev/esbuild/-/esbuild-0.24.0.tgz",
```
Comment (lloydchang) [inline package-lock.json:7610]: **Problem:**

This pull request requires private `npm login` credentials from Roo's AWS CodeArtifact.

**For Example:**

```zsh
npm error code E401
npm error Unable to authenticate, your authentication token seems to be invalid.
npm error To correct this please try logging in again with:
npm error   npm login
npm error A complete log of this run can be found in:  /Users/username/.npm/_logs/2024-11-20T19_54_04_026Z-debug-0.log
```

```
17 silly tarball no local data for esbuild@https://roo-815250993495.d.codeartifact.us-east-1.amazonaws.com/npm/roo-dev/esbuild/-/esbuild-0.24.0.tgz. Extracting by manifest.
```

**Solution:**

I created a different pull request at https://github.com/cline/cline/pull/784 with a fix to undo custom and extraneous changes.

**Review by lloydchang — CHANGES_REQUESTED** [review]
Relates to https://github.com/cline/cline/pull/784#issuecomment-2489454482

**Comment by punkouter25** [comment]
is there an exe for dummys like me to run this version  ?

**Comment by KJ7LNW** [comment]
@punkouter25, see #784 and #802, but to answer your question:

- git clone <repo> cline-dev
- cd cline-dev
- git checkout origin/<branch>
- Open `cline-dev/` in vscode
- Install all suggested plugins, as some of them seem to be required in order to build but I do not know which one(s).
- Press F5 (Run->Start Debugging), and a new vscode instance "[Extension Development Host]" will launch.  
- Use _that new window's_ instance of vscode to open the code you wish to work on with cline-dev. 
- Cleanup:
    - Shift-F5 to stop the debugging instance (from the primary window), or just close the window, and maybe still press Shift-F5 if it gets stuck.

**Comment by lloydchang** [comment]
Alternatively, from https://github.com/cline/cline/pull/784#issuecomment-2495231386
```zsh
gh repo clone cline/cline
cd cline
gh pr checkout 784 --repo cline/cline
npm run install:all
npm install -g @vscode/vsce
vsce package
code --install-extension claude-dev-2.1.6.vsix
code
```
* https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally
* https://code.visualstudio.com/api/working-with-extensions/publishing-extension

**Comment by lloydchang** [comment]
> code --install-extension claude-dev-2.1.6.vsix

@punkouter25 @KJ7LNW @DRieckeBWP

To be clear, `claude-dev-2.1.6.vsix` is the executable in the step-by-step example from https://github.com/cline/cline/pull/674#issuecomment-2495641259

* By installing `claude-dev-2.1.6.vsix`, you no longer need to open two different Visual Studio Code windows — One window for initial, and and a separate window for debug

* Furthermore, that custom `2.1.6` version of the Cline (also known as `claude-dev`) extension is available to new instances of the Visual Studio Code GUI application

**Comment by lloydchang** [comment]
> would like regex for _which_ files or cmds to allow. Just add `.*` if you really want all.

Relates to https://github.com/cline/cline/pull/784#issuecomment-2489496370 and https://github.com/cline/cline/pull/802#issuecomment-2496081701

**Comment by lloydchang** [comment]
Related https://github.com/cline/cline/discussions/368 https://github.com/cline/cline/discussions/635 https://github.com/cline/cline/pull/674 https://github.com/cline/cline/pull/784 https://github.com/cline/cline/pull/802

**Comment by lloydchang** [comment]
if anyone needs Auto Approve Mode, it's available via Roo Cline at https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

**Pull request closed without merging**